### PR TITLE
As a user I should be able to see all the topics in the system

### PR DESCRIPTION
**Pull request by keinydev**
[Trello card here ](https://trello.com/c/cW9M7RnZ/33-1-as-a-user-i-should-be-able-to-see-all-the-topics-in-the-system)

### CODE REVIEW

File: app/controllers/api/v1/topics_controller.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  module V1
+    class TopicsController < ApplicationController
+
+      def index
+        @topics = Topic.all 
+        render json: @topics
+      end         
+    end
+  end
+end
```
Comment (enzofab91) [inline app/controllers/api/v1/topics_controller.rb:11]: Add a line here.

File: app/models/topic.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+#
+# Table name: topics
+#
+#  id         :bigint           not null, primary key
+#  name       :string           not null
+#  image      :string           not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+class Topic < ApplicationRecord
```
Comment (enzofab91) [inline app/models/topic.rb:11]: Maybe here we can add some validation, for example name presence.

See https://guides.rubyonrails.org/active_record_validations.html

File: db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,5 @@
-# This file should contain all the record creation needed to seed the database with its default values.
-# The data can then be loaded with the bin/rails db:seed command (or created alongside the database with db:setup).
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+topic1 = Topic.create(name: "Music",  image: Faker::LoremFlickr.image(search_terms: ['music']))
```
Comment (enzofab91) [inline db/seeds.rb:1]: Nice!

File: spec/requests/api/v1/topics/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+require "rails_helper"
+
+RSpec.describe "POST api/v1/topics", type: :request do
```
Comment (enzofab91) [inline spec/requests/api/v1/topics/show_spec.rb:3]: Remove `Rspec`, I think is unnecessary.

File: Gemfile [hunk 5]
```diff
@@ -34,6 +34,7 @@ group :development, :test do
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
   gem 'rspec-rails', '~> 5.0.0'
   gem 'pry-rails'
+  gem 'faker', :git => 'https://github.com/faker-ruby/faker.git', :branch => 'master'
```
Comment (santir489) [inline Gemfile:37]: Is `:git => 'https://github.com/faker-ruby/faker.git', :branch => 'master'` needed?

File: db/migrate/20210727211310_create_topics.rb [hunk 6]
```diff
@@ -0,0 +1,11 @@
+class CreateTopics < ActiveRecord::Migration[6.1]
+  def change
+    create_table :topics do |t|
+      t.string :name,     null: false
+      t.string :image,    null: false
+      t.timestamps
+    end
+
+    add_index :topics, :name,   unique: true
```
Comment (santir489) [inline db/migrate/20210727211310_create_topics.rb:9]: We should add `null` and `unique` restrictions in the model as well

File: spec/requests/api/v1/topics/show_spec.rb [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+
+  describe 'GET Index' do
+    context 'when the route is valid' do
+      before do
+        get api_v1_topics_path
+      end
+
+      it 'returns status 200' do
+        expect(response).to have_http_status(200)
```
Comment (enzofab91) [inline spec/requests/api/v1/topics/show_spec.rb:12]:
```suggestion
        expect(response).to be_successful
```

**Review by enzofab91 — COMMENTED** [review]
Nice work!

File: spec/rails_helper.rb [hunk 8]
```diff
@@ -34,14 +34,20 @@
   # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
   config.fixture_path = "#{::Rails.root}/spec/fixtures"
 
+  # loading seeds
+  config.before(:suite) do
+    Rails.application.load_seed 
+  end
```
Comment (santir489) [inline spec/rails_helper.rb:40]: Maybe we do not want seed to be loaded in each specs.
I think is better to generate each rspec context in their own file. 
Each test will have specific needs.

File: app/controllers/api/v1/topics_controller.rb (hunk 1 above)
Comment (keinydev) [reply app/controllers/api/v1/topics_controller.rb:11]: Do you mean at the end?

Comment (enzofab91) [reply app/controllers/api/v1/topics_controller.rb:11]: Yes

File: Gemfile (hunk 5 above)
Comment (keinydev) [reply Gemfile:37]: I use it for data generation

File: app/controllers/api/v1/topics_controller.rb (hunk 1 above)
Comment (keinydev) [reply app/controllers/api/v1/topics_controller.rb:11]: Actually there is an extra line at the end of the file

File: spec/requests/api/v1/topics/show_spec.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
+require "rails_helper"
+
+describe "POST api/v1/topics", type: :request do
```
Comment (santir489) [inline spec/requests/api/v1/topics/show_spec.rb:3]: Maybe we can change this,  is not a POST
And
This file should be named `index_spec.rb` right?

File: spec/requests/api/v1/topics/show_spec.rb [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+      before do
+        create_list(:topic, 5)
+        get api_v1_topics_path
+      end
+
+      it 'returns status 200' do
+        expect(response).to be_successful
+      end
+
+      it 'returns data attributes' do
```
Comment (santir489) [inline spec/requests/api/v1/topics/show_spec.rb:16]: It feels to me that we are not testing this properly. 
How do we know that we are returning ALL topics and not only one?

Comment (keinydev) [reply spec/requests/api/v1/topics/show_spec.rb]: Ok

**Review by enzofab91 — APPROVED** [review]

**Review by santir489 — APPROVED** [review]
👏

**Pull request merged**